api4/team.go

```diff
@@ -1581,11 +1581,12 @@ func updateTeamScheme(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	schemeID := model.SchemeIDFromJson(r.Body)
-	if schemeID == nil || (!model.IsValidId(*schemeID) && *schemeID != "") {
+	var p model.SchemeIDPatch
+	if jsonErr := json.NewDecoder(r.Body).Decode(&p); jsonErr != nil || p.SchemeID == nil || (!model.IsValidId(*p.SchemeID) && *p.SchemeID != "") {
 		c.SetInvalidParam("scheme_id")
 		return
 	}
+	schemeID := p.SchemeID
```

Comment:
I find checking the json err and the semantics in the same statement a bit hard to read. Is the following easier to follow?
```suggestion
	var p model.SchemeIDPatch
	if jsonErr := json.NewDecoder(r.Body).Decode(&p); jsonErr != nil {
		c.SetInvalidParam("scheme_id")
		return
	}
	schemeID := p.SchemeID
	
	if schemeID == nil || (!model.IsValidId(*schemeID) && *shemeID != "") { 
		c.SetInvalidParam("scheme_id")
		return
	}
```

Reply:
It's a bit dense I admit. Updated.